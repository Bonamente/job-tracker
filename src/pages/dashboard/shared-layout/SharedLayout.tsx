import { Outlet } from 'react-router-dom';

import { Navbar, BigSidebar, SmallSidebar } from '../../../components';
import StyledSharedLayout from './StyledSharedLayout';

type SharedLayoutProps = {
  switchTheme: () => void;
};

const SharedLayout: React.FC<SharedLayoutProps> = ({ switchTheme }) => {
  return (
    <StyledSharedLayout>
      <div className="dashboard">
        <SmallSidebar />
        <BigSidebar />
        <div>
          <Navbar switchTheme={switchTheme} />
          <main className="dashboard-page">
            <Outlet />
          </main>
        </div>
      </div>
    </StyledSharedLayout>
  );
};

export default SharedLayout;
