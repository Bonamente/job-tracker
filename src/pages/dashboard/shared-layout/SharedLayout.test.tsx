import { renderWithProviders } from 'src/test/test-utils';
import initI18n from 'src/utils/i18n';
import SharedLayout from './SharedLayout';

describe('SharedLayout', () => {
  beforeAll(() => {
    initI18n();
  });

  it('renders SharedLayout', () => {
    const { container } = renderWithProviders(
      <SharedLayout switchTheme={() => {}} />
    );

    expect(container).toMatchSnapshot();
  });
});
