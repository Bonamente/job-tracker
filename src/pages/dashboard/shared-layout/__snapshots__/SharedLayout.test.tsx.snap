// Vitest Snapshot v1

exports[`SharedLayout > renders SharedLayout 1`] = `
<div>
  <div
    class="sc-csuSiG iENGQm"
  >
    <div
      class="dashboard"
    >
      <aside
        class="sc-eDvSVe kTXmYR"
      >
        <div
          class="sidebar-container"
        >
          <div
            class="content"
          >
            <button
              aria-label="close the pop-up navigation menu"
              class="close-btn"
              type="button"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 352 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                />
              </svg>
            </button>
            <header>
              <img
                alt="logo"
                class="logo"
                src="/src/assets/images/logo.svg"
              />
            </header>
            <nav
              class="nav-links"
            >
              <a
                aria-current="page"
                class="nav-link active"
                href="/"
              >
                <span
                  class="icon"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M496 496H16V16h32v448h448v32z"
                    />
                    <path
                      d="M192 432H80V208h112zm144 0H224V160h112zm143.64 0h-112V96h112z"
                    />
                  </svg>
                </span>
                stats
              </a>
              <a
                class="nav-link"
                href="/all-jobs"
              >
                <span
                  class="icon"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M19.88 18.47c.44-.7.7-1.51.7-2.39 0-2.49-2.01-4.5-4.5-4.5s-4.5 2.01-4.5 4.5 2.01 4.5 4.49 4.5c.88 0 1.7-.26 2.39-.7L21.58 23 23 21.58l-3.12-3.11zm-3.8.11a2.5 2.5 0 010-5 2.5 2.5 0 010 5zm-.36-8.5c-.74.02-1.45.18-2.1.45l-.55-.83-3.8 6.18-3.01-3.52-3.63 5.81L1 17l5-8 3 3.5L13 6l2.72 4.08zm2.59.5c-.64-.28-1.33-.45-2.05-.49L21.38 2 23 3.18l-4.69 7.4z"
                    />
                  </svg>
                </span>
                all jobs
              </a>
              <a
                class="nav-link"
                href="/add-job"
              >
                <span
                  class="icon"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 448 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M448 75.2v361.7c0 24.3-19 43.2-43.2 43.2H43.2C19.3 480 0 461.4 0 436.8V75.2C0 51.1 18.8 32 43.2 32h361.7c24 0 43.1 18.8 43.1 43.2zm-37.3 361.6V75.2c0-3-2.6-5.8-5.8-5.8h-9.3L285.3 144 224 94.1 162.8 144 52.5 69.3h-9.3c-3.2 0-5.8 2.8-5.8 5.8v361.7c0 3 2.6 5.8 5.8 5.8h361.7c3.2.1 5.8-2.7 5.8-5.8zM150.2 186v37H76.7v-37h73.5zm0 74.4v37.3H76.7v-37.3h73.5zm11.1-147.3l54-43.7H96.8l64.5 43.7zm210 72.9v37h-196v-37h196zm0 74.4v37.3h-196v-37.3h196zm-84.6-147.3l64.5-43.7H232.8l53.9 43.7zM371.3 335v37.3h-99.4V335h99.4z"
                    />
                  </svg>
                </span>
                add job
              </a>
              <a
                class="nav-link"
                href="/profile"
              >
                <span
                  class="icon"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    version="1.1"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M13.5 0h-12c-0.825 0-1.5 0.675-1.5 1.5v13c0 0.825 0.675 1.5 1.5 1.5h12c0.825 0 1.5-0.675 1.5-1.5v-13c0-0.825-0.675-1.5-1.5-1.5zM13 14h-11v-12h11v12zM4 9h7v1h-7zM4 11h7v1h-7zM5 4.5c0-0.828 0.672-1.5 1.5-1.5s1.5 0.672 1.5 1.5c0 0.828-0.672 1.5-1.5 1.5s-1.5-0.672-1.5-1.5zM7.5 6h-2c-0.825 0-1.5 0.45-1.5 1v1h5v-1c0-0.55-0.675-1-1.5-1z"
                    />
                  </svg>
                </span>
                profile
              </a>
            </nav>
          </div>
        </div>
      </aside>
      <aside
        class="sc-hLBbgP dBapIF"
      >
        <div
          class="sidebar-container show-sidebar"
        >
          <div
            class="content"
          >
            <header>
              <img
                alt="logo"
                class="logo"
                src="/src/assets/images/logo.svg"
              />
            </header>
            <nav
              class="nav-links"
            >
              <a
                aria-current="page"
                class="nav-link active"
                href="/"
              >
                <span
                  class="icon"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 512 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M496 496H16V16h32v448h448v32z"
                    />
                    <path
                      d="M192 432H80V208h112zm144 0H224V160h112zm143.64 0h-112V96h112z"
                    />
                  </svg>
                </span>
                stats
              </a>
              <a
                class="nav-link"
                href="/all-jobs"
              >
                <span
                  class="icon"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M0 0h24v24H0z"
                      fill="none"
                    />
                    <path
                      d="M19.88 18.47c.44-.7.7-1.51.7-2.39 0-2.49-2.01-4.5-4.5-4.5s-4.5 2.01-4.5 4.5 2.01 4.5 4.49 4.5c.88 0 1.7-.26 2.39-.7L21.58 23 23 21.58l-3.12-3.11zm-3.8.11a2.5 2.5 0 010-5 2.5 2.5 0 010 5zm-.36-8.5c-.74.02-1.45.18-2.1.45l-.55-.83-3.8 6.18-3.01-3.52-3.63 5.81L1 17l5-8 3 3.5L13 6l2.72 4.08zm2.59.5c-.64-.28-1.33-.45-2.05-.49L21.38 2 23 3.18l-4.69 7.4z"
                    />
                  </svg>
                </span>
                all jobs
              </a>
              <a
                class="nav-link"
                href="/add-job"
              >
                <span
                  class="icon"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 448 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M448 75.2v361.7c0 24.3-19 43.2-43.2 43.2H43.2C19.3 480 0 461.4 0 436.8V75.2C0 51.1 18.8 32 43.2 32h361.7c24 0 43.1 18.8 43.1 43.2zm-37.3 361.6V75.2c0-3-2.6-5.8-5.8-5.8h-9.3L285.3 144 224 94.1 162.8 144 52.5 69.3h-9.3c-3.2 0-5.8 2.8-5.8 5.8v361.7c0 3 2.6 5.8 5.8 5.8h361.7c3.2.1 5.8-2.7 5.8-5.8zM150.2 186v37H76.7v-37h73.5zm0 74.4v37.3H76.7v-37.3h73.5zm11.1-147.3l54-43.7H96.8l64.5 43.7zm210 72.9v37h-196v-37h196zm0 74.4v37.3h-196v-37.3h196zm-84.6-147.3l64.5-43.7H232.8l53.9 43.7zM371.3 335v37.3h-99.4V335h99.4z"
                    />
                  </svg>
                </span>
                add job
              </a>
              <a
                class="nav-link"
                href="/profile"
              >
                <span
                  class="icon"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    version="1.1"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M13.5 0h-12c-0.825 0-1.5 0.675-1.5 1.5v13c0 0.825 0.675 1.5 1.5 1.5h12c0.825 0 1.5-0.675 1.5-1.5v-13c0-0.825-0.675-1.5-1.5-1.5zM13 14h-11v-12h11v12zM4 9h7v1h-7zM4 11h7v1h-7zM5 4.5c0-0.828 0.672-1.5 1.5-1.5s1.5 0.672 1.5 1.5c0 0.828-0.672 1.5-1.5 1.5s-1.5-0.672-1.5-1.5zM7.5 6h-2c-0.825 0-1.5 0.45-1.5 1v1h5v-1c0-0.55-0.675-1-1.5-1z"
                    />
                  </svg>
                </span>
                profile
              </a>
            </nav>
          </div>
        </div>
      </aside>
      <div>
        <nav
          class="sc-dkrFOg frxxQR"
        >
          <div
            class="nav-center"
          >
            <button
              aria-label="show/hide navigation menu"
              class="toggle-btn"
              title="show/hide navigation menu"
              type="button"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12.83 352h262.34A12.82 12.82 0 0 0 288 339.17v-38.34A12.82 12.82 0 0 0 275.17 288H12.83A12.82 12.82 0 0 0 0 300.83v38.34A12.82 12.82 0 0 0 12.83 352zm0-256h262.34A12.82 12.82 0 0 0 288 83.17V44.83A12.82 12.82 0 0 0 275.17 32H12.83A12.82 12.82 0 0 0 0 44.83v38.34A12.82 12.82 0 0 0 12.83 96zM432 160H16a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16zm0 256H16a16 16 0 0 0-16 16v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16v-32a16 16 0 0 0-16-16z"
                />
              </svg>
            </button>
            <div>
              <img
                alt="logo"
                class="logo"
                src="/src/assets/images/logo.svg"
              />
              <h1
                class="logo-text"
              >
                Dashboard
              </h1>
            </div>
            <div
              class="user-controls"
            >
              <div
                class="sc-gswNZR cIwcWJ"
              >
                <button
                  aria-label="activate dark mode"
                  class="theme-switch-btn btn"
                  title="activate dark mode"
                  type="button"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    style="color: rgb(0, 0, 0);"
                    viewBox="0 0 16 16"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M8 12a4 4 0 1 0 0-8 4 4 0 0 0 0 8zM8 0a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 0zm0 13a.5.5 0 0 1 .5.5v2a.5.5 0 0 1-1 0v-2A.5.5 0 0 1 8 13zm8-5a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2a.5.5 0 0 1 .5.5zM3 8a.5.5 0 0 1-.5.5h-2a.5.5 0 0 1 0-1h2A.5.5 0 0 1 3 8zm10.657-5.657a.5.5 0 0 1 0 .707l-1.414 1.415a.5.5 0 1 1-.707-.708l1.414-1.414a.5.5 0 0 1 .707 0zm-9.193 9.193a.5.5 0 0 1 0 .707L3.05 13.657a.5.5 0 0 1-.707-.707l1.414-1.414a.5.5 0 0 1 .707 0zm9.193 2.121a.5.5 0 0 1-.707 0l-1.414-1.414a.5.5 0 0 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .707zM4.464 4.465a.5.5 0 0 1-.707 0L2.343 3.05a.5.5 0 1 1 .707-.707l1.414 1.414a.5.5 0 0 1 0 .708z"
                    />
                  </svg>
                </button>
              </div>
              <div
                class="sc-bcXHqe kiMojn"
              >
                <button
                  aria-label="Change current language"
                  class="lang-switch-btn btn"
                  type="button"
                >
                  en
                </button>
              </div>
              <div
                class="btn-container"
              >
                <button
                  aria-controls="dropdown"
                  aria-expanded="false"
                  class="btn"
                  type="button"
                >
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 496 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm0 96c48.6 0 88 39.4 88 88s-39.4 88-88 88-88-39.4-88-88 39.4-88 88-88zm0 344c-58.7 0-111.3-26.6-146.5-68.2 18.8-35.4 55.6-59.8 98.5-59.8 2.4 0 4.8.4 7.1 1.1 13 4.2 26.6 6.9 40.9 6.9 14.3 0 28-2.7 40.9-6.9 2.3-.7 4.7-1.1 7.1-1.1 42.9 0 79.7 24.4 98.5 59.8C359.3 421.4 306.7 448 248 448z"
                    />
                  </svg>
                  <span
                    class="user-name"
                  />
                  <svg
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 320 512"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                    />
                  </svg>
                </button>
                <div
                  class="dropdown"
                  id="dropdown"
                >
                  <button
                    class="dropdown-btn"
                    type="button"
                  >
                    sign out
                  </button>
                </div>
              </div>
            </div>
          </div>
        </nav>
        <main
          class="dashboard-page"
        />
      </div>
    </div>
  </div>
  <div
    class="Toastify"
  />
</div>
`;
